napalm_junos/junos.py

```diff
@@ -616,8 +616,7 @@ def get_lldp_neighbors_detail(self, interface=''):
             return {}
         interfaces = lldp_table.get().keys()
 
-        old_junos = napalm_base.helpers.convert(
-            int, self.device.facts.get('version', '0.0').split('.')[0], 0) < 13
+        old_junos = re.search('EX|QFX|J|SRX', self.device.facts.get('model'))
```

Comment:
It's more indicated to check the `personality` field from facts (that regex processing is already well covered in https://github.com/Juniper/py-junos-eznc/blob/master/lib/jnpr/junos/facts/personality.py and it doesn't require regex compilation on method execution).